lib/statsig/api_client.ex

```diff
@@ -1,55 +1,105 @@
+  @default_logging_api_url "https://statsigapi.net/v1"
+  @default_config_specs_api_url "https://api.statsigcdn.com/v1"
+
+  def download_config_specs(since_time \\ 0) do
+    case api_key() do
+      {:ok, key} ->
+        base_url = api_url(@default_config_specs_api_url)
+
+        url =
+          URI.new!(base_url)
+          |> URI.append_path(Path.join("/download_config_specs", key <> ".json"))
+          |> URI.append_query(URI.encode_query(sinceTime: to_string(since_time)))
+          |> URI.to_string()
+
+        case Req.get(url) do
+          {:ok, %Req.Response{status: status, body: %{} = body}} when status in 200..299 ->
+            {:ok, body}
+
+          {:ok, %Req.Response{status: status} = response} when status in 200..299 ->
+            Logger.error("Invalid response format, expected map got: #{inspect(response.body)}")
+            {:error, {:invalid_response_format, response.body}}
+
+          {:ok, %Req.Response{status: status, body: body}} ->
+            Logger.error("HTTP error: status #{status}, body: #{inspect(body)}")
+            {:error, :http_error, status}
+
+          {:error, :unexpected_error, error} ->
+            Logger.error("Unexpected error: #{inspect(error)}")
+            {:error, :unexpected_error, error}
+
+          {:error, error} ->
+            Logger.error("Unexpected error: #{inspect(error)}")
+            {:error, :unexpected_error, error}
+        end
+
+      {:error, reason} ->
+        Logger.error("Failed to get API key: #{reason}")
+        {:error, :missing_api_key, reason}
+    end
+  end
+
+  def push_logs(logs) do
+    case api_key() do
+      {:ok, key} = key_result ->
+        base_url = api_url(@default_logging_api_url)
+
+        url =
+          URI.new!(base_url)
+          |> URI.append_path("/rgstr")
+          |> URI.to_string()
+
+        case Req.post(url: url, json: %{events: logs}, headers: headers(key)) do
+          {:ok, %{status: code}} when code < 300 ->
+            {:ok, []}
+
+          {:ok, response} ->
+            Logger.error(
+              "Failed to push logs: status #{response.status}, body: #{inspect(response.body)}"
+            )
+
+            {:error, logs}
+
+          {:error, error} ->
+            Logger.error("Failed to push logs: #{inspect(error)}")
+            {:error, logs}
+        end
+
+      {:error, reason} ->
+        Logger.error("Failed to get API key: #{reason}")
+        {:error, logs}
     end
   end
 
-  def push_logs(api_key, logs) do
-    base_url = get_api_url(@default_logging_api_url)
-    url = "#{base_url}rgstr"
-    result = Req.post(
-      url: url,
-      json: %{"events" => logs},
-      headers: headers(api_key)
-    ) |> case do
-      {:ok, %{status: code}} when code < 300 -> {:ok, []}
-      _ -> {:error, logs}
+  defp api_key() do
+    case Application.get_env(:statsig, :api_key) do
+      nil ->
+        Logger.error("Statsig API key is not configured")
+        {:error, "API key not configured"}
```

Comment:
it's more idiomatic to have the reasons be atoms. `:api_key_not_configured` would be appropriate here. 
Having atom reasons allows you to define the error tuples as typespecs and enumerate all the different errors. Strings don't let you do that.